www/src/utils/node/docs.js

```diff
@@ -0,0 +1,352 @@
+            node {
+              fields {
+                slug
+                locale
+                package
+                released
+              }
+              frontmatter {
+                title
+                draft
+                canonicalLink
+                publishedAt
+                issue
+                tags
+              }
+            }
+          }
+        }
+      }
+    `).then(result => {
+      if (result.errors) {
+        return reject(result.errors)
+      }
+
+      const blogPosts = _.filter(result.data.allMdx.edges, edge => {
+        const slug = _.get(edge, `node.fields.slug`)
+        const draft = _.get(edge, `node.frontmatter.draft`)
+        if (!slug) return undefined
+
+        if (_.includes(slug, `/blog/`) && !draft) {
+          return edge
+        }
+
+        return undefined
+      })
+
+      const releasedBlogPosts = blogPosts.filter(post =>
+        _.get(post, `node.fields.released`)
+      )
+
+      // Create blog-list pages.
+      const postsPerPage = 8
+      const numPages = Math.ceil(releasedBlogPosts.length / postsPerPage)
+
+      Array.from({
+        length: numPages,
+      }).forEach((_, i) => {
+        createPage({
+          path: i === 0 ? `/blog` : `/blog/page/${i + 1}`,
+          component: slash(blogListTemplate),
+          context: {
+            limit: postsPerPage,
+            skip: i * postsPerPage,
+            numPages,
+            currentPage: i + 1,
+          },
+        })
+      })
+
+      // Create blog-post pages.
+      blogPosts.forEach((edge, index) => {
+        let next = index === 0 ? null : blogPosts[index - 1].node
+        if (next && !_.get(next, `fields.released`)) next = null
+
+        const prev =
+          index === blogPosts.length - 1 ? null : blogPosts[index + 1].node
+
+        createPage({
+          path: `${edge.node.fields.slug}`, // required
+          component: slash(blogPostTemplate),
+          context: {
+            slug: edge.node.fields.slug,
+            prev,
+            next,
+          },
+        })
+      })
+
+      const makeSlugTag = tag => _.kebabCase(tag.toLowerCase())
+
+      // Collect all tags and group them by their kebab-case so that
+      // hyphenated and spaced tags are treated the same. e.g
+      // `case-study` -> [`case-study`, `case study`]. The hyphenated
+      // version will be used for the slug, and the spaced version
+      // will be used for human readability (see templates/tags)
+      const tagGroups = _(releasedBlogPosts)
+        .map(post => _.get(post, `node.frontmatter.tags`))
+        .filter()
+        .flatten()
+        .uniq()
+        .groupBy(makeSlugTag)
+
+      tagGroups.forEach((tags, tagSlug) => {
+        createPage({
+          path: `/blog/tags/${tagSlug}/`,
+          component: tagTemplate,
+          context: {
+            tags,
+          },
+        })
+      })
+
+      // Create docs pages.
+      const docPages = result.data.allMdx.edges
```

Comment:
As a note, there's another PR I made to factor out the prev-and-next logic: #20978